docs/contributor_guide.rst

```diff
@@ -166,21 +193,24 @@ Testing
 -------
 
 Tests should be added to ``taichi/tests``.
-Use ``ti test`` to run all the tests.
-Use ``ti test -v`` for verbose outputs.
-Use ``ti test <filename(s)>`` to run specific tests. e.g. ``ti test numpy_io`` and ``ti test test_numpy_io.py`` are equivalent.
-Use ``ti test -a <arch(s)>`` for test against specified architectures. e.g. ``ti test -a opengl`` or ``ti test numpy_io -a cuda,metal``.
-Use ``ti test -na <arch(s)>`` for test all architectures exclude some of them. e.g. ``ti test -na opengl,cuda``.
-Use ``ti test -c`` to run only the C++ tests. e.g. ``ti test -c alg_simp``
+
+- Use ``ti test`` to run all the tests.
+- Use ``ti test -v`` for verbose outputs.
+- Use ``ti test <filename(s)>`` to run specific tests. e.g. ``ti test numpy_io`` and ``ti test test_numpy_io.py`` are equivalent.
+- Use ``ti test -a <arch(s)>`` for test against specified architectures. e.g. ``ti test -a opengl`` or ``ti test numpy_io -a cuda,metal``.
+- Use ``ti test -na <arch(s)>`` for test all architectures exclude some of them. e.g. ``ti test -na opengl,cuda``.
+- Use ``ti test -c`` to run only the C++ tests. e.g. ``ti test -c alg_simp``
 
 For more options, see ``ti test -h``.
 
 Documentation
 -------------
 
-Use ``ti doc`` to build the documentation locally.
-Open the documentation at ``taichi/doc/build/index.html``.
-On Linux/OS X, use ``watch -n 1 ti doc`` to continuously build the documentation.
+- Use ``ti doc`` to build the documentation locally.
+- Open the documentation at ``taichi/doc/build/index.html``.
+- On Linux/OS X, use ``watch -n 1 ti doc`` to continuously build the documentation.
+
+  - If the OpenGL backend detector keeps creating new windows, execute ``export TI_WITH_OPENGL=0`` for ``ti doc``.
```

Comment:
Can we move these stuffs into `Utilities`, together with `ti benchmark`?

Reply:
Sounds good. I'll open a future PR for that.